src/tr2/game/objects/traps/rolling_ball.c

```diff
@@ -161,11 +162,13 @@ static void M_Collision(
         return;
     }
 
-    if (lara_item->gravity) {
+    if (lara_item->gravity || g_Config.debug.enable_invulnerability) {
         if (coll->enable_baddie_push) {
             Lara_Push(item, coll, coll->enable_hit, true);
         }
-        lara_item->hit_points -= ROLLING_BALL_DAMAGE_AIR;
+        if (!g_Config.debug.enable_invulnerability) {
+            lara_item->hit_points -= ROLLING_BALL_DAMAGE_AIR;
+        }
```

Comment:
We could use `Lara_TakeDamage` TBH, but this can become a part of a bigger cleanup as there are many usages left.
BTW Sometimes Lara receives damage via Item_TakeDamage too. 
Both of these cases are covered by the invulnerability cheat resetting Lara's HP to max in the control routines.